src/ast/int_parser.cpp

```diff
@@ -0,0 +1,140 @@
+#include <algorithm>
+#include <exception>
+#include <sstream>
+#include <stdexcept>
+#include <variant>
+
+#include "int_parser.h"
+
+namespace {
+
+template <typename T>
+T _parse_int(const std::string &num, size_t *idx, int base)
+{
+  throw std::runtime_error("parse_int not implemented for type");
```

Comment:
I think we can make this a compile time error instead. Example: https://godbolt.org/z/sEs3xhac4

Reply:
Ah yeah that works :)